integrations/clickup/client.py

```diff
@@ -0,0 +1,87 @@
+from typing import Any, AsyncGenerator
+
+from httpx import Timeout
+from port_ocean.context.event import event
+from port_ocean.utils import http_async_client
+from port_ocean.utils.cache import cache_iterator_result
+
+TEAMS_CACHE = "teams"
+
+
+class ClickupClient:
+    def __init__(self, clickup_apikey: str) -> None:
+        self.clickup_url = "https://api.clickup.com/api/v2"
+        self.client = http_async_client
+        self.client.headers.update({"Authorization": clickup_apikey})
+        self.client.timeout = Timeout(30)
+
+    async def get_teams(self) -> list[dict[str, Any]]:
+        if cache := event.attributes.get(TEAMS_CACHE, []):
+            return cache
+        return await self._get_teams()
+
+    async def _get_teams(self) -> list[dict[str, Any]]:
+        url = f"{self.clickup_url}/team"
+        response = await self.client.get(url)
+        response.raise_for_status()
+        teams = response.json()["teams"]
+        event.attributes[TEAMS_CACHE] = teams
+        return teams
+
+    @cache_iterator_result()
+    async def get_spaces(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+        for team in await self.get_teams():
+            yield await self._get_spaces(team["id"])
+
+    async def _get_spaces(self, team_id: str) -> list[dict[str, Any]]:
+        url = f"{self.clickup_url}/team/{team_id}/space"
+        response = await self.client.get(url)
+        response.raise_for_status()
+        spaces = response.json()["spaces"]
+        return [{**space, "__team_id": team_id} for space in spaces]
+
+    @cache_iterator_result()
+    async def get_folders(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+        async for spaces in self.get_spaces():
+            for space in spaces:
+                yield await self._get_folders(space["id"], space["__team_id"])
+
+    async def _get_folders(self, space_id: str, team_id: str) -> list[dict[str, Any]]:
+        url = f"{self.clickup_url}/space/{space_id}/folder"
+        response = await self.client.get(url)
+        response.raise_for_status()
+        folders = response.json()["folders"]
+        return [{**folder, "__team_id": team_id} for folder in folders]
```

Comment:
should be const

Reply:
I don't understand your comment though. I assumed you meant "__team_id" should be defined as a constant and called in the 6 places used.